app/controllers/api/v2/courses_controller.rb

```diff
@@ -2,7 +2,7 @@ module API
   module V2
     class CoursesController < ApplicationController
       def index
-        provider = Provider.find_by!(provider_code: params[:provider_code])
+        provider = Provider.friendly.find(params[:provider_id])
```

Comment:
`params[:provider_id]` so is that the `friendly_id` aka `provider_code`?

Reply:
Yes. We define what the slug/url in the model so the controllers and routes are standard rails with the except of the `friendly` helper in the controller to hook it all together.